Open Graph metadata + next on air + roles + FAQ + timeline

dchan3 opened the pull request.

dchan3 committed:
Edit Open Graph meta properties

dchan3 committed:
Playlist sidebar time display fix + now playing bug

dchan3 committed:
Fix next on air & playlist bugs

dchan3 committed:
Change Open Graph properties

dchan3 committed:
Make things better for users with conflicting roles

dchan3 committed:
FAQ adjustment + add timeline to menu

daveynight commented on imports/startup/lib/dj.js:
```diff
@@ -26,8 +26,7 @@ DJ.allow('collections.reviews.showCreate', true);
 
-DJ.deny('collections.reviews.update', (userId, doc, fields, modifier) =>
-  (_.contains(fields, 'approved')));
+DJ.deny('collections.reviews.update', (userId, doc, fields, modifier) => (_.contains(fields, 'approved') && !Meteor.user().hasRole("moderator") && !Meteor.user().hasRole("admin")));
```
This is a very long line of code.

daveynight commented on imports/startup/lib/dj.js:
```diff
@@ -43,8 +42,7 @@ DJ.allow('collections.parties.showCreate', true);
 
-DJ.deny('collections.parties.update', (userId, doc, fields, modifier) =>
-  (_.contains(fields, 'approved')));
+DJ.deny('collections.parties.update', (userId, doc, fields, modifier) => (_.contains(fields, 'approved') && !Meteor.user().hasRole("moderator") && !Meteor.user().hasRole("admin")));
```
This should also be broken up into a 2-3 liner

daveynight commented on imports/startup/lib/dj.js:
```diff
@@ -61,8 +59,7 @@ DJ.allow('collections.posts.showCreate', true);
 
-DJ.deny('collections.posts.update', (userId, doc, fields, modifier) =>
-  (_.contains(fields, 'approved')));
+DJ.deny('collections.posts.update', (userId, doc, fields, modifier) => (_.contains(fields, 'approved') && !Meteor.user().hasRole("moderator") && !Meteor.user().hasRole("admin")));
```
Again, same thing

daveyshindig reviewed (changes requested):
Thanks for all this. I just have a couple nitpicks regarding coding style. If you can amend your commit accordingly, I'll gladly merge. ✅

dchan3 committed:
Implement changes requested in #57

dchan3 committed:
Fix show page playlists bug

dchan3 committed:
Now Playing modified to display sub show info if any


Uncomment stuff in landing


Adjust landing query to shows collection


Revert changes to landing


Fix syntax errors

dchan3 committed:
Review page author display name

dchan3 committed:
Post body links color

dchan3 committed:
Display subbing DJ if necessary

dchan3 committed:
Add show and profile links to landing

dchan3 committed:
Rewrite parties publication query to hide expired events

daveynight commented on imports/ui/components/shows/show_page.html:
```diff
@@ -46,7 +46,7 @@ <h5 class='show__time'>{{day startDay}}s from {{timeBeautify startHour startMinu
         <h5 class='show__latest-header'>Latest Episode</h5>
         <h2 class='show__latest-date'>{{time (latestPlaylist.showDate)}}</h2>
-        {{#if pastPlaylists}}
+        {{#if pastPlaylists.fetch}}
```
Should be `pastPlaylists && pastPlaylists.fetch`, no?

daveyshindig commented on imports/ui/components/shows/show_page.js:
```diff
@@ -39,11 +40,12 @@ Template.showPage.helpers({
   pastPlaylists: () => Playlists.find({}, { sort: { showDate: -1 }, skip: 1}),
   playlistsByYear: () => {
-    var playlistDates = Playlists.find({}, {sort: {showDate: -1}, skip: 1}).fetch();
+    var playlistDates = Playlists.find({ showId: Shows.findOne({ slug: FlowRouter.getParam('slug') }).showId }, { sort: { showDate: -1 }, skip: 1 }).fetch();
```
Again, this needs to be at least two or three lines.

daveynight replied:
Please break this up into 2-3 lines.

daveynight commented on imports/api/playlists/server/publications.js:
```diff
@@ -16,6 +16,17 @@ Meteor.publish('playlistsLimited', function(options) {
 
+Meteor.publish('currentPlaylist', function() {
+  return Playlists.find({$where: function() {
+    return this.showDate.getYear() === new Date().getYear() &&
```
I'm seeing this function a few times. Can we factor it out into a separate function elsewhere in the project? This will improve maintainability.

daveynight commented on imports/ui/components/home/landing.js:
```diff
@@ -31,6 +36,31 @@ Template.landing.helpers({
+                              startHour: { $lte: now.getHours() }, endDay: now.getDay(),
+                              endHour: { $gt: now.getHours() } });
+    var playlist = Playlists.findOne({$where: function() {
+      return this.showDate.getYear() === new Date().getYear() &&
```
Yup, let's factor this out since it's being repeated elsewhere.

daveynight commented on imports/api/parties/server/publications.js:
```diff
@@ -2,7 +2,7 @@ import { Meteor } from 'meteor/meteor';
 
 Meteor.publish('approvedParties', function () {
-  return Parties.find({approved: true});
+  return Parties.find({ approved: true, startTime: { $gt: new Date() } });
```
Commits like this are too small. Please combine small commits like this in the future. You can do that through creating a new branch and doing an interactive rebase.

dchan3 committed:
Implement more requested changes in #57

dchan3 committed:
Correct errors relating to shows and users

daveynight commented on imports/ui/components/home/landing.js:
```diff
@@ -24,6 +24,19 @@ Template.landing.onCreated(function() {
+    }
+  }
+)};
+
```
This appears in another file, correct? My intention was to see this factored out into a utility/helper file, since it appeared in separate files

dchan3 replied:
It's been factored out into `imports/startup/lib/helpers.js`.

dchan3 committed:
Show DJ link bug + current playlist helper + profile link in landing

dchan3 committed:
Configure email  + delete resend queue

dchan3 committed:
Implement resend email verification + default member role for new users

dchan3 committed:
Improve parties create/update datetime input

daveynight commented on imports/api/playlists/server/publications.js:
```diff
@@ -16,16 +17,7 @@ Meteor.publish('playlistsLimited', function(options) {
-           this.endTime === "00:00:00"); }
-  });
-});
+Meteor.publish('currentPlaylist', currentPlaylist);
```
💯

daveynight commented on imports/ui/components/resend/resend.js:
```diff
@@ -23,5 +23,23 @@ Template.resend.events({
+      default:
+        if ($('.validate').val().match(/[A-Za-z0-9.]+@([a-z0-9.])+\.[a-z]{2,3}/) !== null) {
+          if ($(".submit").prop("disabled")) {
+            $(".submit").prop("disabled", false);
```
I reflexively dislike this construct. Can't we use a single `if` clause here?

daveynight commented on imports/ui/components/resend/resend.js:
```diff
@@ -23,5 +23,23 @@ Template.resend.events({
+          }
+        }
+        else {
+          if (!$(".submit").prop("disabled")) {
```
`else if` is a thing in JS, no?

dchan3 committed:
Implement more changes requested in #57

daveyshindig reviewed (approved):
Chee pono
